packages/shared/src/contexts/payment/StoreKit.tsx

```diff
@@ -79,63 +137,29 @@ export const StoreKitSubProvider = ({
   const { user, isValidRegion: isPlusAvailable } = useAuthContext();
   const { logSubscriptionEvent } = usePlusSubscription();
   const { growthbook } = useGrowthBookContext();
-  const productIds = useFeature(featureIAPProducts);
-  const productList = useMemo(() => Object.keys(productIds), [productIds]);
   const logRef = useRef<typeof logSubscriptionEvent>();
   logRef.current = logSubscriptionEvent;
 
+  const { data } = useQuery<ProductPricingMetadata[]>({
+    queryKey: generateQueryKey(RequestKey.PricePreview, user, 'ios', 'plus'),
+    queryFn: () => fetchPricingMetadata(ProductPricingType.Plus),
+    enabled: !!user && !!growthbook?.ready && iOSSupportsPlusPurchase(),
+    staleTime: StaleTime.Default,
+  });
+
   const { data: productOptions } = useQuery({
     queryKey: ['iap-products'],
-    enabled: !!productIds && !!growthbook?.ready && iOSSupportsPlusPurchase(),
+    enabled: !!data?.length && !!growthbook?.ready && iOSSupportsPlusPurchase(),
+    staleTime: StaleTime.Default,
     queryFn: async () => {
       if (!messageHandlerExists(WebKitMessageHandlers.IAPSubscriptionRequest)) {
         return [];
       }
 
-      const response = promisifyEventListener<
-        ProductOption[],
-        IAPProduct[] | string
-      >('iap-products-result', (event) => {
-        const productsRaw = !isNullOrUndefined(event?.detail)
-          ? event.detail
-          : [];
-
-        // Remove JSON parsing once usage of App v1.8 is low
-        const products: IAPProduct[] =
-          typeof productsRaw === 'string'
-            ? JSON.parse(productsRaw)
-            : productsRaw;
-
-        return products
-          ?.map((product: IAPProduct): ProductOption => {
-            const { duration, label, extraLabel, appsId } =
-              productIds[product.attributes.offerName];
-
-            return {
-              label,
-              value: product.attributes.offerName,
-              price: {
-                amount: parseFloat(product.attributes.offers[0].price),
-                formatted: product.attributes.offers[0].priceFormatted,
-              },
-              extraLabel,
-              appsId: appsId ?? PlusPriceTypeAppsId.Default,
-              duration,
-              durationLabel:
-                duration === PlusPriceType.Yearly ? 'year' : 'month',
-              trialPeriod: null,
-            };
-          })
-          .sort((a: { value: string }, b: { value: string }) => {
-            // Make sure that the products are sorted in the same order as the product list
```

Comment:
This still sorts somewhere else?

Reply:
We can sort it now based on the order in our metadata, and if we are going to do some sorting here, we should also sort it in all other places for consistency. I'll make sure the product team will know how to properly use the Retool app 🙏